src/cpp/common/alarm.cc

```diff
@@ -74,9 +84,18 @@ class AlarmImpl : public grpc::internal::CompletionQueueTag {
         [this] { OnCQAlarm(absl::OkStatus()); });
   }
   void Set(gpr_timespec deadline, std::function<void(bool)> f) {
+    if (queued_.load() != executed_.load()) {
```

Comment:
Consider making this a helper method instead of duplicating the code. This will grow when backoff logic gets added.

Reply:
done.